### PR TITLE
Add Save as PDF option

### LINKED ISSUE #1284: Add Save as PDF option to export the processed image
Note: **Please note we are preparing to participate in Google Code-in, and have reserved this issue for GCI participants - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more.**

### Save as PDF
In order to save maximum printable quality of the processed image, SAVE AS PDF option should be added along with other Export options.

##### See the image below!
Screenshot 2019-10-28 at 10 35 41 PM

### PR DESCRIPTION
**Pull request by anthony-zhou**
Fixes #1284 

I added a "Save as PDF" option for export, using the dependency jsPDF, which I installed in the project. Here is a screenshot of the options box:

image

Here is a screenshot of the PDF that was downloaded (I named the file "index.pdf" -- is that a good filename for now or is there something more descriptive we should change it to?):

image

Here is an example of a PDF generated with the new "Save as PDF" option: 

image

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below
* [x] Insert-step functionality is working correct as expected.

### CODE REVIEW

**Comment by anthony-zhou** [comment]
@publiclab/is-reviewers How does this change look?

File: examples/demo.js [hunk 1]
```diff
@@ -207,6 +209,39 @@ window.onload = function () {
...
+  //returns the data URL for the last image in the sequence
+  function getLastImage() {
+    //get the image of the last step
+    let imgs = document.getElementsByClassName('step-thumbnail');
+    let lastStepImage = imgs[imgs.length-1];
+    return lastStepImage.getAttribute("src");
+  }
+
+  /*
+  * savePDF()
```
Comment (HarshKhandeparkar) [inline examples/demo.js]: Could you please convert this to a [JSDoc comment](https://devdocs.io/jsdoc/)?

File: examples/demo.js [hunk 2]
```diff
@@ -207,6 +209,39 @@ window.onload = function () {
     });
   }
 
+  //returns the data URL for the last image in the sequence
```
Comment (HarshKhandeparkar) [inline examples/demo.js]: Please start with a capital letter, end with a `.` and add space between `//` and the sentence. We are trying to maintain consistency among all comments. Thank you.

**Comment by anthony-zhou** [comment]
@HarshKhandeparkar Thanks for the feedback! I have edited the function comments to fit JSDoc comment guidelines, and the inline comments to start with a capital letter and end with a period. Is this the format you were looking for?

File: examples/demo.js [hunk 3]
```diff
@@ -207,6 +209,44 @@ window.onload = function () {
...
+  */
+  function getLastImage() {
+    // Get the image from the last step.
+    let imgs = document.getElementsByClassName('step-thumbnail');
+    let lastStepImage = imgs[imgs.length-1];
+    return lastStepImage.getAttribute("src");
+  }
+
+  /**
+  * Download the given image URL as a PDF file.
```
Comment (SidharthBansal) [inline examples/demo.js]: Remove full stop from phrases.

**Comment by SidharthBansal** [comment]
Do we actually need https://github.com/publiclab/image-sequencer/pull/1349/files#diff-32607347f8126e6534ebc7ebaec4853d in this pr?
Thanks both of you. I am not mentoring IS.
I think @keshav234156 can help you
Thanks

File: examples/demo.js (hunk 3 above)
Comment (HarshKhandeparkar) [reply examples/demo.js]: Isn't it a complete sentence?

Comment (anthony-zhou) [reply examples/demo.js]: It seems like the [JSDoc examples](https://devdocs.io/jsdoc/about-block-inline-tags) usually look like this (they use imperative sentences).

Comment (HarshKhandeparkar) [inline examples/demo.js]: I think you should use the `@description` directive instead of direct comment.
```js
* @description Download.....
```

**Comment by jywarren** [comment]
This looks great. Thank you so much!

**Pull request merged**